webapp/src/blocks.tsx

```diff
@@ -705,6 +707,21 @@ export class Editor extends toolboxeditor.ToolboxEditor {
         }
     }
 
+    async validateTutorialCode(tutorial: pxt.tutorial.TutorialOptions) {
+        // Current tutorial step
+        const { tutorialStepInfo, tutorialStep } = tutorial;
+        const step = tutorialStepInfo[tutorialStep];
+
+        // Get the current XML and call the validator function
+        const source = this.getCurrentSource();
+        // Get current TypeScript as well, for comarison
+        const file = pkg.mainEditorPkg().files["main.ts"].content;
+
+        let blocks = this.editor.getAllBlocks();
+
+        this.parent.setTutorialCodeStatus(tutorialStep, await validator.validate(step, source, blocks, this.blockInfo));
```

Comment:
This would be a good place to catch any kind of bugs in the validation code (and report them to our analytics)